File: ecommerce-shop-ui/src/app/common/product-info.ts
import { Category } from './category';
import { Image } from './image';
import { Product } from './product';
import { Stock } from './stock';

export class ProductInfo {

    product : Product;
    stock : Stock;
    category : Category;
    mainProductImage : Image;
    imgList : Image[];

    constructor(product: Product, stock: Stock, category: Category, mainProductImage : Image, images : Image[]){
        this.product = product;
        this.stock = stock;
        this.category = category;
        this.mainProductImage = mainProductImage;
        this.imgList = images;
    }

}

File: Product-Catalog-Service/src/main/java/com/ryoliveira/ecommerce/productcatalogservice/model/ProductInfo.java
package com.ryoliveira.ecommerce.productcatalogservice.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ProductInfo {

    private Product product;
    private Stock stock;
    private Category category;
    private Image mainProductImage;
    private List<Image> imgList;

}
